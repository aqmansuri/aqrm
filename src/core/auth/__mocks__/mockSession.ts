import { Session } from '../../models';

export const mockSession: Session = {
  token: '3884915llf950',
  user: {
    id: 42,
    firstName: 'Joe',
    lastName: 'Tester',
    email: 'test@test.org',
  },
};
