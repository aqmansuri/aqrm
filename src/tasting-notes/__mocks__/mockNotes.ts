import { TastingNote } from '../../shared/models';

export const mockNotes: TastingNote[] = [
  {
    id: 4,
    brand: 'Lipton',
    name: 'Yellow Label',
    notes: 'Overly acidic, highly tannic flavor',
    rating: 4,
    teaCategoryId: 1,
  },
  {
    id: 73,
    brand: 'Bently',
    name: 'Brown Label',
    notes: 'Essentially OK',
    rating: 3,
    teaCategoryId: 2,
  },
  {
    id: 42,
    brand: 'Lipton',
    name: 'Yellow Label',
    notes: 'Overly acidic, highly tannic flavor',
    rating: 1,
    teaCategoryId: 3,
  },
];
