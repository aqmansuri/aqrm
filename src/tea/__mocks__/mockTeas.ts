import { Tea } from '../../shared/models';

export const expectedTeas = [
  {
    id: 1,
    name: 'Green',
    image: require(`../../assets/images/green.jpg`).default,
    description: 'Green tea description.',
    rating: 0,
  },
  {
    id: 2,
    name: 'Black',
    image: require(`../../assets/images/black.jpg`).default,
    description: 'Black tea description.',
    rating: 1,
  },
  {
    id: 3,
    name: 'Herbal',
    image: require(`../../assets/images/herbal.jpg`).default,
    description: 'Herbal Infusion description.',
    rating: 2,
  },
  {
    id: 4,
    name: 'Oolong',
    image: require(`../../assets/images/oolong.jpg`).default,
    description: 'Oolong tea description.',
    rating: 3,
  },
  {
    id: 5,
    name: 'Dark',
    image: require(`../../assets/images/dark.jpg`).default,
    description: 'Dark tea description.',
    rating: 4,
  },
  {
    id: 6,
    name: 'Puer',
    image: require(`../../assets/images/puer.jpg`).default,
    description: 'Puer tea description.',
    rating: 5,
  },
  {
    id: 7,
    name: 'White',
    image: require(`../../assets/images/white.jpg`).default,
    description: 'White tea description.',
    rating: 0,
  },
  {
    id: 8,
    name: 'Yellow',
    image: require(`../../assets/images/yellow.jpg`).default,
    description: 'Yellow tea description.',
    rating: 0,
  },
];

export const resultTeas = () => {
  return expectedTeas.map((t: Tea) => {
    const tea = { ...t };
    // @ts-ignore
    delete tea.image;
    delete tea.rating;
    return tea;
  });
};
