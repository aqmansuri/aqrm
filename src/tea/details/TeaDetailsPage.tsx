import {
  IonBackButton,
  IonButtons,
  IonContent,
  IonHeader,
  IonPage,
  IonTitle,
  IonToolbar,
  IonImg,
} from '@ionic/react';
import { useState, useEffect } from 'react';
import { useParams } from 'react-router';
import { Rating } from '../../shared/components';
import { Tea } from '../../shared/models';
import { useTea } from '../useTea';

const TeaDetailsPage: React.FC = () => {
  const { id } = useParams<{ id: string }>();
  const { getTeaById, saveTea } = useTea();
  const [tea, setTea] = useState<Tea | undefined>(undefined);

  useEffect(() => {
    (async () => {
      const tea = await getTeaById(parseInt(id, 10));
      setTea(tea);
    })();
  }, [id, getTeaById]);

  return (
    <IonPage>
      <IonHeader>
        <IonToolbar>
          <IonButtons slot="start">
            <IonBackButton defaultHref="/tabs/tea" />
          </IonButtons>
          <IonTitle>Details</IonTitle>
        </IonToolbar>
      </IonHeader>
      <IonContent className="main-content">
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">Details</IonTitle>
          </IonToolbar>
        </IonHeader>
        <div className="ion-padding">
          <div className="ion-justify-content-center">
            <IonImg src={tea?.image} />
          </div>
          <h1>{tea?.name}</h1>
          <p>{tea?.description}</p>
          <Rating
            initialRating={tea?.rating}
            disabled={!tea}
            onRatingChange={rating => saveTea({ ...tea!, rating })}
          />
        </div>
      </IonContent>
    </IonPage>
  );
};
export default TeaDetailsPage;
