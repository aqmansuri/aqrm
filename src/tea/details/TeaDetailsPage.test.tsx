import { render, waitFor } from '@testing-library/react';
import TeaDetailsPage from './TeaDetailsPage';
import { expectedTeas } from '../__mocks__/mockTeas';

jest.mock('react-router', () => ({
  useParams: () => ({
    id: 1,
  }),
}));

const mockTea = expectedTeas[0];
jest.mock('../useTea', () => ({
  useTea: () => ({
    getTeas: jest.fn(),
    getTeaById: jest.fn(() => Promise.resolve(mockTea)),
  }),
}));

describe('<TeaDetailsPage />', () => {
  it('displays the header', async () => {
    const { container } = render(<TeaDetailsPage />);
    await waitFor(() => expect(container).toHaveTextContent(/Details/));
  });

  it('renders consistently', async () => {
    const { asFragment } = render(<TeaDetailsPage />);
    await waitFor(() => expect(asFragment()).toMatchSnapshot());
  });

  it('renders the tea name', async () => {
    const { container } = render(<TeaDetailsPage />);
    await waitFor(() => expect(container).toHaveTextContent(mockTea.name));
  });

  it('renders the tea description', async () => {
    const { container } = render(<TeaDetailsPage />);
    await waitFor(() =>
      expect(container).toHaveTextContent(mockTea.description),
    );
  });
});
