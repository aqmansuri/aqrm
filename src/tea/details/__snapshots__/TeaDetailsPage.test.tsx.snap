// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeaDetailsPage /> renders consistently 1`] = `
<DocumentFragment>
  <div
    class="ion-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-buttons
          slot="start"
        >
          <ion-back-button
            default-href="/tea"
          />
        </ion-buttons>
        <ion-title>
          Details
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content>
      <ion-header
        collapse="condense"
      >
        <ion-toolbar>
          <ion-title
            size="large"
          >
            Details
          </ion-title>
        </ion-toolbar>
      </ion-header>
      <div
        class="ion-padding"
      >
        <div
          class="ion-justify-content-center"
        >
          <ion-img />
        </div>
        <h1 />
        <p />
        <div
          class="rating"
          style="opacity: 0.25;"
        >
          <ion-icon
            data-testid="Rate 1 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          />
          <ion-icon
            data-testid="Rate 2 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          />
          <ion-icon
            data-testid="Rate 3 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          />
          <ion-icon
            data-testid="Rate 4 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          />
          <ion-icon
            data-testid="Rate 5 stars"
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Star</title><path d='M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
          />
        </div>
      </div>
    </ion-content>
  </div>
</DocumentFragment>
`;
